### ISSUE TITLE
mjs and cjs extensions not detected by default

### ISSUE THREAD

**Issue by ericcornelissen** [depcheck/depcheck#733]
### Bug Description

depcheck does not seem to recognize JavaScript files using the file extensions `.mjs` and `.cjs` by default. These file extensions are used to allow writing JavaScript with ESModule syntax in CommonJS projects (`.mjs`) or JavaScript with CommonJS syntax in ESModule projects (`.cjs`). You can read more about this in the Node.js documentation, see [ECMAScript modules](https://nodejs.org/api/esm.html).

I would expect depcheck to recognize files using `.mjs` and `.cjs` by default, rather then having to configure it along the lines of:

```json5
{
  // Other configuration...
  "parsers": {
    "**/*.js": "es6",
    "**/*.cjs": "es6",
    "**/*.mjs": "es6"
  }
}
```

This problem was raised before in #615 and #669, but has not been addressed beyond recommendations to use the configuration.

### Code snippets

See https://github.com/ericcornelissen/depcheck-bugs/tree/e29920d7d63b14b3825a9f83301a7726dd312dbc/mjs-and-cjs _(the specific folder in the repository)_ for a minimum project reproducing the problem.

#### 1. `.mjs`

Where the dependency is used:

```cjs
// ./index.mjs

import test from 'ava';

test('my passing test', t => {
	t.pass();
});
```

Where the dependency is listed in package.json:

```json5
{
  "name": "depcheck-with-mjs-and-cjs",
  "version": "0.0.1",
  "description": "Example of depcheck not detecting dependencies used in .mjs and .cjs files",
  "private": true,
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "vet": "depcheck"
  },
  "author": "Eric Cornelissen",
  "license": "Unlicense",
  "dependencies": {
    "ava": "4.3.0", // <--- Here
    "dotenv": "16.0.1"
  },
  "devDependencies": {
    "depcheck": "1.4.3"
  },
  "engine": {
    "node": "^14.18.0 || ^16.0.0 || ^18.0.0"
  }
}
```

#### 1. `.cjs`

Where the dependency is used:

```cjs
// ./index.cjs

require("dotenv").config();

console.log(`Hello ${process.env.name}!`);
```

Where the dependency is listed in package.json:

```json5
{
  "name": "depcheck-with-mjs-and-cjs",
  "version": "0.0.1",
  "description": "Example of depcheck not detecting dependencies used in .mjs and .cjs files",
  "private": true,
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "vet": "depcheck"
  },
  "author": "Eric Cornelissen",
  "license": "Unlicense",
  "dependencies": {
    "ava": "4.3.0",
    "dotenv": "16.0.1" // <--- Here
  },
  "devDependencies": {
    "depcheck": "1.4.3"
  },
  "engine": {
    "node": "^14.18.0 || ^16.0.0 || ^18.0.0"
  }
}
```

### Versions

- `node -v`: v14.17.6
- `npm -v`: 6.14.15
- `depcheck --version`: 1.4.3

### Extra info

Output of `depcheck --json`

```json
{
  "dependencies": [
    "ava",
    "dotenv"
  ],
  "devDependencies": [],
  "missing": {},
  "using": {
    "depcheck": [
      "/mnt/236BF4A2278DBC3C/workspace/_tmp/depcheck-bugs/mjs-and-cjs/package.json"
    ]
  },
  "invalidFiles": {},
  "invalidDirs": {}
}
```

**Comment by jlarmstrongiv** [comment]
For the cli version, `--parsers='**/*.js:es6,**/*.cjs:es6,**/*.mjs:es6'`

**Comment by mdaverde** [comment]
> For the cli version, `--parsers='**/*.js:es6,**/*.cjs:es6,**/*.mjs:es6'`

Seems like specifying parsers are a complete replacement rather than a granular override of the default parsers. Had to copy and paste the ones in [constants.js](https://github.com/depcheck/depcheck/blob/main/src/constants.js). Looks like a nice workaround for now though!

**Comment by karlhorky** [comment]
> Seems like specifying parsers are a complete replacement rather than a granular override of the default parsers. Had to copy and paste the ones in [constants.js](https://github.com/depcheck/depcheck/blob/main/src/constants.js). Looks like a nice workaround for now though!

@mdaverde What does this mean? Are you implying that the `--parsers='**/*.js:es6,**/*.cjs:es6,**/*.mjs:es6'` as written in @jlarmstrongiv's [comment above](https://github.com/depcheck/depcheck/issues/733#issuecomment-1170141802) does not work? If so, what is your working version that you copied from `constants.js`?

**Comment by majo44** [comment]
For api call you can use:
```
const depcheck = require('depcheck');
    const config = {
        parsers: {
            // as by default cjs/mjs files are not processed we will define the parsers by self
            '**/*.cjs': depcheck.parser.jsx,
            '**/*.js': depcheck.parser.jsx,
            '**/*.mjs': depcheck.parser.jsx,
            '**/*.ts': depcheck.parser.typescript,
            '**/*.tsx': depcheck.parser.typescript,
            '**/*.sass': depcheck.parser.sass,
            '**/*.scss': depcheck.parser.sass,
        },
        ...
    };
depcheck(cwd(), config).then(...);
```

**Comment by karlhorky** [comment]
Hmm... is it possible that `depcheck` does indeed support `.cjs` and `.mjs` files out of the box? It's at least reading [our `stylelint.config.cjs`](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/blob/main/stylelint.config.cjs) file... 🤔

**Comment by ericcornelissen** [comment]
> What does this mean? Are you implying that the `--parsers='**/*.js:es6,**/*.cjs:es6,**/*.mjs:es6'` as written in [the earlier comment] does not work? If so, what is your working version that you copied from constants.js?

What was meant is that that configuration works insofar as it makes depecheck parse `.cjs` and `.mjs` files but is limited in that non-`.js`, `.mjs`, or `.cjs` files may no longer be parsed correctly.


> Hmm... is it possible that depcheck does indeed support `.cjs` and `.mjs` files out of the box? (without configuration) It's at least reading [our stylelint.config.cjs](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/blob/main/stylelint.config.cjs) file... 🤔

Interesting find, I'm guessing you mean to say depcheck doesn't complain about [stylelint dependencies](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/blob/e744c7489ffa23cc6601ae8be5547e726d62dd10/package.json#L69-L72) due to the stylelint configuration file? If so, just to make sure, does depcheck complain about stylelint dependencies when the configuration file is not there? If it does complain, my guess would be that it's a _configuration_ file and those are handled differently from _source_ files.

**Comment by karlhorky** [comment]
## Workaround

Since https://github.com/depcheck/depcheck/pull/773 has not yet been published (and may not be for some time), here's a workaround using configuration, inspired by @jlarmstrongiv and @ericcornelissen (thanks!):

```bash
depcheck --parsers="**/*.js:jsx,**/*.mjs:jsx,**/*.cjs:jsx,**/*.jsx:jsx,**/*.graphql:graphql,**/*.cts:typescript,**/*.mts:typescript,**/*.ts:typescript,**/*.tsx:typescript,**/*.sass:sass,**/*.scss:sass,**/*.vue:vue,**/*.svelte:svelte"
```

This configures [all parsers in `src/constants.js`](https://github.com/depcheck/depcheck/blob/8ead447f7a65cb3f24a9f487395ca6132b1dfc88/src/constants.js#L49-L68) except for [the CoffeeScript parsers](https://github.com/depcheck/depcheck/blob/8ead447f7a65cb3f24a9f487395ca6132b1dfc88/src/constants.js#L54-L56) and [the Storybook and `tsconfig.json` parsers](https://github.com/depcheck/depcheck/blob/8ead447f7a65cb3f24a9f487395ca6132b1dfc88/src/constants.js#L58-L59)
